### PR TITLE
Lint and format docs

### LINKED ISSUE #135: Lint and format docs
With the documentation pages living in the repository, we can enable linters and formatters for the markdown files.

I think we can use:
- [mdformat](https://github.com/executablebooks/mdformat) (I tend to use prettier in other projects, but this is a good alternative to stick with Python)
- [codespell](https://github.com/codespell-project/codespell)

_Aside, we can also add [pre-commit hooks](https://pre-commit.com) for the codebase to improve overall developer experience._

### PR DESCRIPTION
**Pull request by pavithraes**
Fixes #135

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -42,8 +42,12 @@ lint-cpp:
 	# clang-format --dry-run -Werror -i -style=file `find ./cpp/ -name "*.*pp"`
 	echo "C++ linting disabled for now"
 
+lint-docs:
+	mdformat docs/wiki/ README.md examples/README.md
```
Comment (timkpaine) [inline Makefile:46]: `lint-docs` should be:
- `python -m mdformat --check`
- `python -m codespell_lib <omit --write>`

Then seperately, add `fix-docs` and add to `fix` target.

**Review by timkpaine — CHANGES_REQUESTED** [review]
Can you edit the commit message from `Re-push updates with signed commits` to something descriptive of the change itself, e.g. `Add docs linting/fixing`.

**Comment by pavithraes** [comment]
Thanks for the review, I've made the necessary updates :)

File: pyproject.toml [hunk 2]
```diff
@@ -59,6 +59,8 @@ develop = [
     "isort>=5,<6",
     "ruamel.yaml",
     "ruff>=0.3,<0.4",
+    "mdformat",
```
Comment (timkpaine) [inline pyproject.toml:62]: these probably need to be added to the `dev-environment.yml` conda env too @ngoldbaum

**Review by timkpaine — CHANGES_REQUESTED** [review]

**Comment by ngoldbaum** [comment]
It doesn’t exist already but it might be nice to have a linting subsection in the dev guide, explaining what is linted with what tools and with a pointer toward `make fix`.

**Comment by timkpaine** [comment]
> It doesn’t exist already but it might be nice to have a linting subsection in the dev guide, explaining what is linted with what tools and with a pointer toward `make fix`.

Feel free to add more to the [existing lint and fix section](https://github.com/Point72/csp/blob/main/docs/wiki/98.-Building-From-Source.md#lint-and-autoformat)

**Comment by ngoldbaum** [comment]
Oops, forgot that was in the build instructions.

**Comment by timkpaine** [comment]
Also the [contributing guidelines](https://github.com/Point72/csp/blob/main/docs/wiki/98.-Building-From-Source.md#contributing) should be moved to the 99 developer page as they include required information (like e.g. closing PRs with non-substantive changes or without prior discussion for big features). But we can do in a followup PR to avoid conflicts

**Review by timkpaine — DISMISSED** [review]
3 followup changes:

- [x] Add these 2 new deps to conda yaml
- [x] Move contributing guidelines from 98 to 99
- [x] Expand 98 linting/fixing docs and link from 99

**Review by ngoldbaum — APPROVED** [review]

**Pull request merged**